### PR TITLE
TST: clip inplace doesn't modify views with CoW

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/copy_view/test_clip.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+def test_clip_inplace_reference(using_copy_on_write):
+    df = DataFrame({"a": [1.5, 2, 3]})
+    df_copy = df.copy()
+    arr_a = get_array(df, "a")
+    view = df[:]
+    df.clip(lower=2, inplace=True)
+
+    # Clip not actually inplace right now but could be
```
Comment (phofl) [inline pandas/tests/copy_view/test_clip.py:15]: Can you do: TODO(CoW): ...

Comment (phofl) [reply pandas/tests/copy_view/test_clip.py:15]: Ah no, could you add another test without a reference? I think it should pass after my where pr is in

Comment (lithomas1) [reply pandas/tests/copy_view/test_clip.py:15]: I think it is not actually inplace even without CoW.

Comment (phofl) [reply pandas/tests/copy_view/test_clip.py:15]: Yep you are correct. We should refactor this if we decide to keep inplace here

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @lithomas1

**Pull request merged**